libraries/rush-lib/src/cli/actions/InstallAction.ts

```diff
@@ -59,7 +63,10 @@ export class InstallAction extends BaseInstallAction {
       // it is safe to assume that the value is not null
       maxInstallAttempts: this._maxInstallAttempts.value!,
       // These are derived independently of the selection for command line brevity
-      filteredProjects: Array.from(await this._selectionParameters!.getSelectedProjectsAsync(this._terminal)),
+      filteredProjects:
+        selectedProjects?.size === this.rushConfiguration.projects.length
+          ? []
+          : Array.from(selectedProjects ?? []),
```

Comment:
Why is `filteredProjects` typed as an array in the first place? It's internal and the callers start with a `Set`.